clang/include/clang/Sema/Sema.h

```diff
@@ -12963,10 +12963,26 @@ void Sema::addIntelSYCLSingleArgFunctionAttr(Decl *D,
       return;
     }
     int32_t ArgInt = ArgVal->getSExtValue();
-    if (ArgInt <= 0) {
-      Diag(E->getExprLoc(), diag::err_attribute_requires_positive_integer)
-          << CI.getAttrName() << /*positive*/ 0;
-      return;
+    if (CI.getParsedKind() == ParsedAttr::AT_SYCLIntelNumSimdWorkItems ||
+        CI.getParsedKind() == ParsedAttr::AT_IntelReqdSubGroupSize) {
+      if (ArgInt <= 0) {
+        Diag(E->getExprLoc(), diag::err_attribute_requires_positive_integer)
+            << CI.getAttrName() << /*positive*/ 0;
```

Comment:
```suggestion
            << CI.getAttrName() << /*non-negative*/ 1;
```

Reply:
Not related to your patch, but since you are here...

Reply:
Not a problem. I added that codes. Updated now.

Reply:
Oh, that was a wrong suggestion, sorry.